src/deploy_amm.cairo

```diff
@@ -0,0 +1,151 @@
+        let voladjspd_eth_call_lpt: felt252 = 15; // TODO check, no increase??
+        let voladjspd_eth_put_lpt: felt252 =
+            25000; // TODO check, no increase?? // also BTC put pool
+        // 1 BTC voladjspd for btc call pool
+
+        let mut state = Governance::unsafe_new_contract_state();
+        assert(!state.proposal_initializer_run.read(45), 'prop already initialized');
+        state.proposal_initializer_run.write(45, true);
+
+        let governance_addr = get_contract_address();
+        let usdc_addr: ContractAddress = USDC_ADDRESS.try_into().unwrap();
+        let eth_addr: ContractAddress = ETH_ADDRESS.try_into().unwrap();
+        let btc_addr: ContractAddress = BTC_ADDRESS.try_into().unwrap();
+
+        let mut amm_calldata = ArrayTrait::<felt252>::new();
+        amm_calldata.append(governance_addr.into());
+        let amm_deploy_retval = deploy_syscall(amm_class, 42, amm_calldata.span(), false);
+        let (amm_addr, _) = amm_deploy_retval.unwrap_syscall();
+        state.amm_address.write(amm_addr);
+        let amm = IAMMDispatcher { contract_address: amm_addr };
+
+        let eth_call_lpt_addr = deploy_lptoken(
+            amm,
+            'Carmine ETH/USDC call pool',
+            'C-ETHUSDC-C',
+            usdc_addr,
+            eth_addr,
+            0,
+            FixedTrait::from_unscaled_felt(voladjspd_eth_call_lpt)
+        );
+        let eth_put_lpt_addr = deploy_lptoken(
+            amm,
+            'Carmine ETH/USDC put pool',
+            'C-ETHUSDC-P',
+            usdc_addr,
+            eth_addr,
+            1,
+            FixedTrait::from_unscaled_felt(voladjspd_eth_put_lpt)
+        );
+        let btc_call_lpt_addr = deploy_lptoken(
+            amm,
+            'Carmine BTC/USDC call pool',
+            'C-BTCUSDC-C',
+            usdc_addr,
+            btc_addr,
+            0,
+            FixedTrait::ONE()
+        );
+        let btc_put_lpt_addr = deploy_lptoken(
+            amm,
+            'Carmine BTC/USDC put pool',
+            'C-BTCUSDC-P',
+            usdc_addr,
+            btc_addr,
+            1,
+            FixedTrait::from_unscaled_felt(voladjspd_eth_put_lpt)
+        );
+
+        set_trading_halt_permissions(amm);
+
+        Options::add_1201_options(
+            eth_call_lpt_addr, eth_put_lpt_addr, btc_call_lpt_addr, btc_put_lpt_addr
+        );
+    }
+
+    fn deploy_lptoken(
+        amm: IAMMDispatcher,
+        name: felt252,
+        symbol: felt252,
+        quote_token_address: ContractAddress,
+        base_token_address: ContractAddress,
+        option_type: OptionType,
+        voladjspd: Fixed
+    ) -> ContractAddress {
+        let lptoken_class: ClassHash = LP_TOKEN_CLASS_HASH.try_into().unwrap();
+        let mut lpt_calldata: Array<felt252> = ArrayTrait::<felt252>::new();
+        lpt_calldata.append('Carmine ETH/USDC call pool');
+        lpt_calldata.append('C-ETHUSDC-C');
```

Comment:
shouldn't these two add name and symbol?

Reply:
yes, thanks